CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JMultiChoiceComboBoxMenuNGTest.java

```diff
@@ -0,0 +1,369 @@
+        assertEqualsNoOrder(itemsbeforeRemove.toArray(), selectedItems);
+
+        instance.removeSelectedItem(s2);
+        Set resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 2);
+        assertEqualsNoOrder(resultSet.toArray(), new String[]{s1, s3});
+    }
+
+    /**
+     * Test of removeSelectedItems method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testRemoveSelectedItems() {
+        System.out.println("removeSelectedItems");
+        String s1 = "Item 1";
+        String s2 = "Item 3";
+        String s3 = "Item 4";
+        String[] selectedItems = {s1, s2, s3};
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+
+        instance.addSelectedItems(s1, s2, s3);
+        Set itemsbeforeRemove = instance.getSelectedItems();
+        assertEquals(itemsbeforeRemove.size(), 3);
+        assertEqualsNoOrder(itemsbeforeRemove.toArray(), selectedItems);
+
+        instance.removeSelectedItems(s2, s3);
+        Set resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 1);
+        assertEquals(resultSet.toArray(), new String[]{s1});
+    }
+
+    /**
+     * Test of clearSelection method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testClearSelection() {
+        System.out.println("clearSelection");
+        String s1 = "Item 1";
+        String s2 = "Item 3";
+        String s3 = "Item 4";
+        String[] selectedItems = {s1, s2, s3};
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+
+        instance.addSelectedItems(s1, s2, s3);
+        Set itemsbeforeRemove = instance.getSelectedItems();
+        assertEquals(itemsbeforeRemove.size(), 3);
+        assertEqualsNoOrder(itemsbeforeRemove.toArray(), selectedItems);
+
+        instance.clearSelection();
+        Set resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 0);
+    }
+
+    /**
+     * Test of addSelectionListener method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testAddSelectionListener() {
+        System.out.println("addSelectionListener");
+        ListSelectionListener listSelectionListenerMock = mock(ListSelectionListener.class);
+        JMultiChoiceComboBoxMenu instance = mock(JMultiChoiceComboBoxMenu.class);
+        instance.addSelectionListener(listSelectionListenerMock);
+        verify(instance, times(1)).addSelectionListener(listSelectionListenerMock);
+    }
+
+    /**
+     * Test of valueChanged method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testValueChanged() {
+        System.out.println("valueChanged");
+        JMultiChoiceComboBoxMenu instance = mock(JMultiChoiceComboBoxMenu.class);
+        ListSelectionEvent mockEvent = mock(ListSelectionEvent.class);
+        instance.valueChanged(mockEvent);
+        verify(instance, times(1)).valueChanged(mockEvent);
```

Comment:
possibly same as above here

Reply:
So how would you verify the value is actually changed here?

Reply:
I'd assign some listener mocks to the instance's `listeners` then call `valueChanged()`. Then verify that the method iterated over all listeners and called `valueChanged()` on them.

Reply:
Thanks Aldebaran. Fixed now.